docs/src/components/header/ThemeModeToggle.tsx

```diff
@@ -1,18 +1,62 @@
 import * as React from 'react';
+import { useColorScheme, useTheme } from '@mui/material/styles';
 import IconButton from '@mui/material/IconButton';
 import Tooltip from '@mui/material/Tooltip';
 import DarkModeOutlined from '@mui/icons-material/DarkModeOutlined';
 import LightModeOutlined from '@mui/icons-material/LightModeOutlined';
 
-interface ThemeModeToggleProps {
+const CssVarsModeToggle = (props: { onChange: (checked: boolean) => void }) => {
+  const [mounted, setMounted] = React.useState(false);
+  const { mode, setMode } = useColorScheme();
+  React.useEffect(() => {
+    setMounted(true);
+  }, []);
+  return (
+    <Tooltip title={mode === 'dark' ? 'Turn on the light' : 'Turn off the light'}>
+      <IconButton
+        color="primary"
+        disableTouchRipple
+        disabled={!mode}
+        onClick={() => {
+          props.onChange(mode === 'light');
+          setMode(mode === 'dark' ? 'light' : 'dark');
+        }}
+      >
+        {!mode || !mounted
+          ? null
+          : {
+              light: <DarkModeOutlined fontSize="small" />,
+              system: <DarkModeOutlined fontSize="small" />,
+              dark: <LightModeOutlined fontSize="small" />,
+            }[mode]}
+      </IconButton>
+    </Tooltip>
+  );
+};
+
+export default function ThemeModeToggle(props: {
   checked: boolean;
   onChange: (checked: boolean) => void;
-}
-
-export default function ThemeModeToggle(props: ThemeModeToggleProps) {
+}) {
+  const theme = useTheme();
+  if (theme.vars) {
+    // Temporaly renders conditionally because `useColorScheme` could not be used in the pages that haven't migrated to CSS theme variables.
+    return <CssVarsModeToggle {...props} />;
+  }
   return (
     <Tooltip title={props.checked ? 'Turn on the light' : 'Turn off the light'}>
-      <IconButton color="primary" disableTouchRipple onClick={() => props.onChange(!props.checked)}>
+      <IconButton
+        color="primary"
+        disableTouchRipple
+        onClick={() => {
+          try {
+            localStorage.setItem('mui-mode', props.checked ? 'light' : 'dark');
+          } catch (error) {
+            // do nothing
+          }
```

Comment:
The local storage in an event handler, why not have it done in an effect? It feels like this logic is already handled by `docs/src/modules/components/AppSettingsDrawer.js`.

Reply:
> The local storage in an event handler
Setting local storage in this case is an event performed by the user, not an Effect. https://beta.reactjs.org/learn/you-might-not-need-an-effect

> It feels like this logic is already handled by docs/src/modules/components/AppSettingsDrawer.js.
`AppSettingsDrawer` lives in documentation pages, but not marketing pages.

Reply:
> AppSettingsDrawer lives in documentation pages, but not marketing pages.

Great point, what about the same logic that is in `docs/src/modules/components/ThemeContext.js`?

Reply:
@siriwatknp Could you see if it works with?

```suggestion
```

The logic in `docs/src/layouts/AppHeader.tsx` seems to already do the job.